games/game_witcher3.py

```diff
@@ -11,20 +15,32 @@ class Witcher3Game(BasicGame):
 
     Name = "Witcher 3 Support Plugin"
     Author = "Holt59"
-    Version = "1.0.0a"
-    Description = "The Description Of The Dead"
+    Version = "1.1.0"
+    Description = "The Description Of The Dead| Updated to work with GOG by Luca/EzioTheDeadPoet"
 
     GameName = "The Witcher 3: Wild Hunt"
     GameShortName = "witcher3"
     GaneNexusHame = "witcher3"
     GameNexusId = 952
     GameSteamId = 292030
     GameBinary = "bin/x64/witcher3.exe"
-    GameDataPath = "Mods"
+    GameDataPath = ""
     GameSaveExtension = "sav"
     GameDocumentsDirectory = "%DOCUMENTS%/The Witcher 3"
     GameSavesDirectory = "%GAME_DOCUMENTS%/gamesaves"
 
+    # GOG Compatibility
+    def isInstalled(self):
```

Comment:
This breaks the steam detection. You need to call `super().isInstalled()` at some point, either before checking for GOG or after.

Reply:
How would I go about that ? 
This might be a very stupid question but I am not that familiar with python, I mostly used this reference [here](https://github.com/Al12rs/modorganizer-game_generic/blob/master/game_darkestdungeon.py#L363) and just tweaked it, at first trying to write a Titanfall 2 plugin (it worked from the MO side of things but Origin somehow breaks the MO functionality) and then thought I can use it to add GOG compatibility for the Witcher 3 plugin. 


I will try if I get a version of the plugin that launches and has that steam issue hopefully fixed by myself, if it does I would need someone to test if Steam detection works again. But if you could give me an explanation how to do it I am open to do it as well.

Another Idea:
 Doesn't 
```GameSteamId = 292030```
cause the plugin to make a similar check by checking the same place in the registry every time just with a different Steam ID ?

So if this check would also be added for example
```GameGogID = 1495134320``` 
Plugin authors would just need to search for the GOG ID of a game in their registry at ```HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\GOG.com\Games\``` and write it down there and the game could be automatically be detected. (it might be difficult for plugin authors who don't have a GOG version but for that they could ask the community for the ID of the game.)

Reply:
You can change the code to something like this:

```python
def isInstalled(self):
    if super().isInstalled():
        return True  # If Steam installation found, return.

    try:
        RawKey = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, "SOFTWARE\\WOW6432Node\\GOG.com\\Games\\1495134320")
        Key = winreg.QueryValueEx(RawKey, "path")
        winreg.CloseKey(RawKey)
        Dir = re.search("'(.*)'", str(Key))
        self.setGamePath(str(Dir[1]))
        return True
    except:
        return False
```

If you want to check for GOG before Steam, you can `return super().isInstalled()` in the `except` block.

Reply:
> Doesn't `GameSteamId = 292030` cause the plugin to make a similar check by checking the same place in the registry every time just with a different Steam ID ?

The Steam check is slightly different because the path is not stored in the registry. If what you're suggesting is to add an auto-detection feature based on a `GameGogID` attribute, it's probably doable. 

I added Steam first because MO2 only has real "support" for Steam, but other stores work, so why not. I tried adding Origin at some point but I did not find a way to auto-detect games so...

Reply:
Yeah origin has no way of consistency regarding this but the way the launcher is starting its games, or at least Titanfall 2 makes it ignore the vfs making a plugin for it useless.